packages/jupyter-ai/jupyter_ai/extension.py

```diff
@@ -76,39 +78,66 @@ def initialize_settings(self):
         # consumers a Future that resolves to the Dask client when awaited.
         dask_client_future = loop.create_task(self._get_dask_client())
 
+        eps = entry_points()
         # initialize chat handlers
+        self.log.info("Found entry point groups " + ", ".join(sorted(eps.groups)))
+        chat_handler_eps = eps.select(group="jupyter_ai.chat_handlers")
+
+        jai_chat_handlers = {}
+
         chat_handler_kwargs = {
             "log": self.log,
             "config_manager": self.settings["jai_config_manager"],
             "root_chat_handlers": self.settings["jai_root_chat_handlers"],
+            # Everything below this line is the union of all arguments used to
+            # instantiate chat handlers.
+            "chat_history": self.settings["chat_history"],
+            "root_dir": self.serverapp.root_dir,
+            "dask_client_future": dask_client_future,
```

Comment:
Third-parties developing their own chat handlers may want access to these attributes, and there's no reason to restrict them from doing so. Once we work out how we deal with the learn & ask chat handlers, we can make the following changes:

1. Declare these args in `BaseChatHandler.__init__()`
2. Delete the args from all the other chat handler `__init__()` methods.